lib/src/utils.dart

```diff
@@ -148,6 +149,17 @@ Future<List<T>> waitAndPrintErrors<T>(Iterable<Future<T>> futures) {
   });
 }
 
+/// Returns a [StreamTransformer] that will call [onData] for each data
+/// event of the stream.
+///
+/// The stream will be passed through unchanged.
+StreamTransformer<T, T> onDataTransformer<T>(void Function(T data) onData) {
```

Comment:
Can't this be done with `await for`?

Reply:
So, Sigurd [suggested that as well](https://github.com/dart-lang/pub/pull/3546#discussion_r974138761), but I made a mistake when first trying it (forgot to yield chunks).

After adding that in, it seems to work. I'll test and make the change.